xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererSoftware.cpp

```diff
@@ -99,32 +99,38 @@ void CRendererSoftware::RenderImpl(CD3DTexture& target, CRect& sourceRect, CPoin
 
   int swsFlags = SWS_BILINEAR | SWS_PRINT_INFO;
 
-  // Hack to make swscale use its fully-featured scaling algorithms for full chroma interpolation.
+  // Chroma interpolation problems of libswscale as of 7.1.100 / ffmpeg 6.0.1:
+  // - 8 bit source and destination: swscale uses a "special converter" limited to nearest
+  // neighbor for chroma. It's used when source size = dest size and filter lengths are 1.
+  //   > to avoid, set an identity filter of length > 1
+  // - AV_PIX_FMT_X2BGR10 destination: full chroma interpolation is not available and there is no
+  //  swscale API to retrieve the presence of support.
+  //   > no workaround
   //
-  // ff_get_unscaled_swscale() uses nearest neighbor for chroma and is used when source size = dest size
-  // and all filter lengths are 1.
-  // To avoid it, set an identity filter of length > 1
-  //
-  // As of libswscale 7.1.100 / ffmpeg 6.0.1, full chroma interpolation is not available with
-  // AV_PIX_FMT_X2BGR10 destination, and there is no swscale API to retrieve support availability.
-  if (!m_srcFilter && dstFormat == AV_PIX_FMT_BGRA)
-  {
-    m_srcFilter = sws_getDefaultFilter(0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0);
-
-    sws_freeVec(m_srcFilter->lumH);
-    SwsVector* vec = sws_allocVec(3);
-    vec->coeff[0] = 0.0;
-    vec->coeff[1] = 1.0;
-    vec->coeff[2] = 0.0;
-    m_srcFilter->lumH = vec;
+  // - 10 bit source / 8 bit dest: "normal" scaler is used by default, with good results.
 
+  if (dstFormat == AV_PIX_FMT_BGRA)
+  {
     swsFlags |= SWS_FULL_CHR_H_INT;
+
+    if (!m_srcFilter && (buf->av_format == AV_PIX_FMT_YUV420P || buf->av_format == AV_PIX_FMT_NV12))
+    {
+      m_srcFilter = sws_getDefaultFilter(0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0);
+
+      sws_freeVec(m_srcFilter->lumH);
+      SwsVector* vec = sws_allocVec(3);
+      vec->coeff[0] = 0.0;
+      vec->coeff[1] = 1.0;
+      vec->coeff[2] = 0.0;
+      m_srcFilter->lumH = vec;
+    }
   }
 
   // 1. convert yuv to rgb
-  m_sw_scale_ctx = sws_getCachedContext(m_sw_scale_ctx, buf->GetWidth(), buf->GetHeight(),
-                                        buf->av_format, buf->GetWidth(), buf->GetHeight(),
-                                        dstFormat, swsFlags, m_srcFilter, nullptr, nullptr);
+  m_sw_scale_ctx =
+      sws_getCachedContext(m_sw_scale_ctx, buf->GetWidth(), buf->GetHeight(), buf->av_format,
+                           buf->GetWidth(), buf->GetHeight(), dstFormat, swsFlags,
+                           swsFlags & SWS_FULL_CHR_H_INT ? m_srcFilter : nullptr, nullptr, nullptr);
```

Comment:
```suggestion
                           m_srcFilter, nullptr, nullptr);
```
this logic is redundant because filter should be used always when is not nullptr (and when is null is allowed pass nullptr anyway).